cpp/src/Ice/Buffer.cpp

```diff
@@ -90,6 +90,11 @@ IceInternal::Buffer::Container::operator=(Container&& other) noexcept
 {
     if (this != &other)
     {
+        // If we own the buffer, free it first.
+        if (_buf && _owned)
```

Comment:
Looks good!

Side note: It’s fine to call `::free(nullptr)`. I think we should refactor the `::free` calls in this class to remove the `_buf` check.